src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerations/AvoidMultipleEnumerations.cs

```diff
@@ -178,15 +178,22 @@ private void OnOperationBlockStart(OperationBlockStartAnalysisContext operationB
             }
 
             var syntaxTree = operationBlocks[0].Syntax.SyntaxTree;
-            var customizedEnumerationMethods = operationBlockStartAnalysisContext.Options.GetEnumerationMethodsOption(
+            var options = operationBlockStartAnalysisContext.Options;
+            var customizedEnumerationMethods = options.GetEnumerationMethodsOption(
                     MultipleEnumerableDescriptor,
                     syntaxTree,
-                    operationBlockStartAnalysisContext.Compilation);
+                    compilation);
 
-            var customizedLinqChainMethods = operationBlockStartAnalysisContext.Options.GetLinqChainMethodsOption(
+            var customizedLinqChainMethods = options.GetLinqChainMethodsOption(
                     MultipleEnumerableDescriptor,
                     syntaxTree,
-                    operationBlockStartAnalysisContext.Compilation);
+                    compilation);
+
+            var assumeMethodEnumeratesArguments = options.GetAssumeMethodEnumeratesArgumentsOption(
+                MultipleEnumerableDescriptor,
+                syntaxTree,
+                compilation,
+                defaultValue: false);
```

Comment:
Tag @mikadumont , this is the default value I talked to you offlilne.

Reply:
Since main is in dotnet 7 preview, I feel the default value won't be a blocking issue for this analyzer. Even if we want to change it we could do it in a later preview.
I would like to merge the feature branch back to main after this PR is checked in. So that this analyzer itself could be shipped first.